## Building Biomedical Knowledge Graphs

1. Set up the context for relationship extraction
	1. Define the problem
	2. Talk about the importance of the problem (filling knowledge bases -> point researchers to relevant papers)
2. Give overview towards taxonomy of approaches (hand written rules, unsupervised machine learning, supervised machine learning etc.)

### Constructing Databases and Manual Curation

Constructing a database is an extensive process that can date back to the early 1990s [@pmid:7584356].
This process involves gathering relevant text such as journal articles, abstracts, or web-based text.
Each collection of text undergoes the process of tagging and normalizing important keywords to represent entities of interest.
This task is commonly known as named entity recognition. 
Typically, this task uses a dictionary look-up approach [@doi:10.1093/database/bas037] or a machine learning system [@doi:10.1093/bioinformatics/btw343; @doi:10.1142/9789812776136_0062; @doi:10.1186/1471-2105-12-S8-S5] to identify the keywords of interest.
Following named entity recognition, the next step is to detect sentences that assert a relationship of interest (i.e. relationship extraction).
This task can be accomplished via manual curation, automated approaches or a combination of both.
Succeeding detection, identified sentences are incorporated as evidence for establishing a relationship between mentioned entities of interest. 

Notable databases have been constructed using an extensive amount of manual curation (Table {@tbl:manual-curated-databases}).
For example, COSMIC [@doi:10.1093/nar/gkw1121] was constructed via a group of domain experts scanning the literature for key cancer related genes.
This database has reached close to 45k entries in 2016 and continues to grow throughout the years [@doi:10.1093/nar/gkw1121].
Studies have shown that these databases contain relatively precise data, but in low quantities [@doi:10.1038/nmeth1209-860; @doi:10.1038/nmeth.1284; @doi:10.1093/database/bau058].
This happens because the high publication rate is too much for curators to keep up [@doi:10.1007/s11192-010-0202-z].
Plus, this process isn't completely error free as databases were found to be missing annotations and entries [@doi:10.1093/bioinformatics/btm229].
These combined findings highlight a significant need for faster and more accurate approaches.

There have been attempts to fix this bottleneck via semi-automatic ways [@doi:10.1016/j.jbi.2010.11.001; @doi:10.1093/database/bau067; @doi:10.1093/database/bas010; @doi:10.1093/database/baz068].
One approach compared annotating PubMed queries with and without the help of an automated system [@doi:10.1016/j.jbi.2010.11.001].
It was found that annotating with pre-annotated queries provided a better speed up than annotating from scratch.
Despite the speed up, there was some bias found in the corrected annotations.
In cases where a query could fall into multiple categories, most annotators defaulted to the same single category.
This bias highlights a reoccurring problem of ambiguous sentences.
Some sentences can have multiple implications, which makes manual curation a difficult task.

Manual curation is still need for relationship extraction approaches.
This process can be used to generate gold standard datasets that automated systems use for calibration [@doi:10.1016/j.jbi.2012.04.004; @doi:10.1016/j.artmed.2004.07.016].
Plus, approaches can use semi-automatic methods to speed up this arduous process.
Nevertheless, constructing databases using only manual curation is not sufficient.
Databases constructed this way will have a significant amount of missing information, since there is too much data to humanly process.
Future applications should focus on more automated methods to improve the quality of relationship extraction.

| Database [Reference] | Short Description |
| --- | --- |
| Entrez-Gene [@doi:10.1093/nar/gki031] | NCBI's Gene annotation database that contains information pertaining to genes, gene's organism source, phenotypes etc. | 
| Swiss-Prot [@doi:10.1093/nar/gkg095] | A protein protein interaction database that contains proteomic information. |
| PharmGKB [@doi:10.1093/nar/30.1.163] | A database that contains genetic, phenotypic, and clinical information related to pharmacogenomic studies. |
| COSMIC [@doi:10.1093/nar/gkw1121] | A database that contains high resolution human cancer genetic information. | 
| BioGrid [@doi:10.1093/nar/gks1158] | A database for major model organisms. It contains genetic and proteomic information.| 
| Comparative Toxicogenomics Database [@doi:10.1186/1471-2105-10-326] | A database that contains manually curated chemical-gene-disease interactions and  relationships. |
| [@doi:10.1093/nar/gkr1149] | A death domain superfamily database. | 
| Comprehensive Antibiotic Resistance Database [@doi:10.1093/nar/gkw1004] | Manually curated database that contains information about the molecular basis of antimicrobial resistance.|
Table. A table of reputable databases that were populated through extensive use of manual curation.{#tbl:manual-curated-databases}


### Text Mining for Relationship Extraction

#### Rule-Based Natural Language Processing

1. Mention papers on hand written rules and expressions

#### Unsupervised Machine Learning

1. Mention Clustering Analysis
2. Mention Co-Occurrence approaches

#### Supervised Machine Learning

1. Mention the availablility of publically available data
	1. PPI - 5 datasets 
	   1. 10.1016/j.artmed.2004.07.016 
	   2. 10.1186/1471-2105-8-50 
	   3. Learning language in logic - genic interaction extraction challenge
	   4. 10.1093/bioinformatics/btl616 
	   5. http://helix-web.stanford.edu/psb02/ding.pdf
	2. DaG - 3 datasets
	   1. 10.1016/j.jbi.2012.04.004 
	   2. 10.1186/s12859-015-0472-9
	   3. 10.1186/1471-2105-14-323 
	   4. 10.1186/1471-2105-13-161
	3. CiD 
	  1. 10.1093/database/baw068 
	4. CbG 
	  1. Biocreative VI track 5 - raw citation
	5. more if exists talk about deep learning methods
2. Mention the use of Support Vector Machines and other non deep learning classifiers
   1. Will have to mention that field has moved to deep learning.
   2. 10.1186/s13326-017-0168-3
   3. 10.1371/journal.pcbi.1004630
3. Mention deep learning methods
   1. 1901.06103v1
   2. 10.1016/j.knosys.2018.11.020
   3. 10.1177/0165551516673485
   4. 1706.01556v2
   5. ^^ A few papers here but a lot more will be put into place 
   6. Mention caveat which is the need for large annotated datasets
   7. Mention a direction the field is moving to which is weak supervision and more that info that will come in time.
