## Building Biomedical Knowledge Graphs

1. Set up the context for relationship extraction
	1. Define the problem
	2. Talk about the importance of the problem (filling knowledge bases -> point researchers to relevant papers)
2. Give overview towards taxonomy of approaches (hand written rules, unsupervised machine learning, supervised machine learning etc.)

### Constructing Databases and Manual Curation

1. Talk about papers that construct knowledge graphs without text mining approaches
2. Discuss the positives and negatives for these methods

### Text Mining for Relationship Extraction

#### Rule-Based Natural Language Processing

1. Mention papers on hand written rules and expressions

#### Unsupervised Machine Learning

Unsupervised machine learning consists of drawing inferences within data without the use of labels.
Common methods of unsupervised learning involve some form of clustering or statistical calculations.
In this section we discuss methods that use unsupervised learning to detect relationship asserting sentences from text.

Majority of unsupervised text mining methods use some form of co-occurrence (Table {@tbl:unsupervised-methods-text-mining}), which is an instance of two entities appearing together in text.
Two popular databases DISEASES [@doi:10.1016/j.ymeth.2014.11.020] and STRING [@doi:10.1093/nar/gku1003] were populated using a co-occurrence scoring method on PubMed abstracts.
This scoring method measured the frequency of co-mention pairs both within individual sentences and within the abstracts itself.
Mention pairs that occur more than chance were assumed to have a high likelihood of association or interaction.
This approach was able to recall a significant amount of relationships, but had a hard time capturing relationships that are rare or underreported.
 
Besides PubMed abstracts, a study that used a similar approach to mine full articles for protein-protein interactions and other protein related information [@doi:10.1371/journal.pcbi.1005962].
It was discovered that using full text significantly improved detection performance compared to abstracts alone.
This improvement suggests that future text mining approaches should consider using full text to increase detection power.

Following co-occurrence, one study took clustering approach to detect relevant sentences. 
This approach consisted of using a blclustering algorithm on generated dependency parse trees to group PubMed abstract sentences together.
Each cluster was manually curated to determine which relationship each cluster represented [@doi:10.1093/bioinformatics/bty114].
This approach recalled a significant amount of pre-existing relationships and generated 36 different groups.
Despite the success, this approach suffered from technical issues.
Dependency trees are prone to parsing errors, which results in sentences being lost by the clustering algorithm.
Based on this flaw, rare or under represented relationships could be missed.

Overall unsupervised methods provide a means to rapidly find relationship asserting sentences without the need of annotated text.
From co-occurrence and clustering methods there has been much success, but  most of these methods need to an abundance of sentences in order to be detected.
Future unsupervised methods should focus on improving power to detect these rare sentences and relationships.

| Study | Relationship of Interest | 
| --- | --- | 
| [@doi:10.1093/bioinformatics/btz490] | Protein-Protein Interactions, Disease-Gene and Tissue-Gene Associations |
| [@doi:10.1109/bibm.2015.7359766] | Drug Disease Treatments |
| [@doi:10.1371/journal.pcbi.1000943] | Drug, Gene and Disease interactions |
| [@doi:10.1371/journal.pcbi.1005962]| Protein-Protein Interactions |
| [@doi:10.1016/j.ymeth.2014.11.020] | Disease-Gene associations|
| [@doi:10.1093/nar/gku1003]| Protein-Protein Interactions |
| [@doi:10.1371/journal.pcbi.1005017] | Genotype-Phenotype Relationships |

Table: Table of approaches that mainly use a form of co-occurrence. {#tbl:unsupervised-methods-text-mining}

#### Supervised Machine Learning

1. Mention the availablility of publically available data
	1. PPI - 5 datasets 
	   1. 10.1016/j.artmed.2004.07.016 
	   2. 10.1186/1471-2105-8-50 
	   3. Learning language in logic - genic interaction extraction challenge
	   4. 10.1093/bioinformatics/btl616 
	   5. http://helix-web.stanford.edu/psb02/ding.pdf
	2. DaG - 3 datasets
	   1. 10.1016/j.jbi.2012.04.004 
	   2. 10.1186/s12859-015-0472-9
	   3. 10.1186/1471-2105-14-323 
	   4. 10.1186/1471-2105-13-161
	3. CiD 
	  1. 10.1093/database/baw068 
	4. CbG 
	  1. Biocreative VI track 5 - raw citation
	5. more if exists talk about deep learning methods
2. Mention the use of Support Vector Machines and other non deep learning classifiers
   1. Will have to mention that field has moved to deep learning.
   2. 10.1186/s13326-017-0168-3
   3. 10.1371/journal.pcbi.1004630
3. Mention deep learning methods
   1. 1901.06103v1
   2. 10.1016/j.knosys.2018.11.020
   3. 10.1177/0165551516673485
   4. 1706.01556v2
   5. ^^ A few papers here but a lot more will be put into place 
   6. Mention caveat which is the need for large annotated datasets
   7. Mention a direction the field is moving to which is weak supervision and more that info that will come in time.
