## Applying Knowledge Graphs to Biomedical Challenges

1. Mention that these graphs can be used for discovery
2. Mention representation learning (aka representing a graph as dense vectors for nodes and/or edges)
3. 

### Unifying Techniques

Mapping high dimensional data into a low dimensional space has greatly improved modeling performance in fields such as natural language processing [@arxiv:1301.3781; @arxiv:1310.4546] and image analysis [@arxiv:1512.03385].
The success of these approaches provides rationale for projecting knowledge graphs into a low dimensional space as well [@arxiv:1709.05584].
Techniques that perform this projection often require information on how nodes are connected with one another [@raw:gongreplearn; @doi:10.1109/TKDE.2016.2606098; @raw:hanreplearn; @doi:10.1145/2806416.2806512], while other approaches can work directly with the edges themselves [@arxiv:1610.04073].
We group methods for producing low-dimensional representations of knowledge graphs into the following three categories: matrix factorization, translational methods, and deep learning.

#### Matrix Factorization

1. Mention techniques for these with some papers

#### Translational Distance Models

Translational distance models treat edges in a knowledge graph as linear transformations.
As an example, one such algorithm, TransE [@raw:bordestranse], treats every node-edge pair as a triplet with head nodes represented as $\textbf{h}$, edges represented as $\textbf{r}$, and tail nodes represented as $\textbf{t}$.
These representations are combined into an equation that mimics the iconic word vectors translations ($\textbf{king} - \textbf{man} + \textbf{woman} \approx \textbf{queen}$) from the Word2vec model [@arxiv:1310.4546].
The equation is shown as follows: $\textbf{h} + \textbf{r} \approx \textbf{t}$.
Starting at the head node ($\textbf{h}$), add the edge vector ($\textbf{r}$) and the result should be the tail node ($\textbf{t}$).
TransE optimizes embeddings for $\textbf{h}$, $\textbf{r}$, $\textbf{t}$, while guaranteeing the global equation ($\textbf{h} + \textbf{r} \approx \textbf{t}$) is satisfied [@raw:bordestranse].
A caveat to the TransE approach is that it the training steps force relationships to have a one to one mapping, which may not be appropriate for all types of relationships.

Wang et al. [@raw:wangtransH] attempted to resolve the one to one mapping issue by developing the TransH model.
TransH treats relations as hyperplanes rather than a regular vector and projects the head ($\textbf{h}$) and tail ($\textbf{t}$) nodes onto the hyperplane.
Following this projection, a distance vector ($\textbf{d}_{r}$) is calculated between the projected head and tail nodes.
Finally, each vector is optimized while preserving the global equation ($\textbf{h} + \textbf{d}_{r} \approx \textbf{t}$) [@raw:wangtransH].
Other approaches [@raw:lintransR, @arxiv:1610.04073; @arxiv:1909.00672] have built off of the TransE and TransH models. 
In the future, it may be beneficial for these models is to incorporate other types of information such as edge confidence scores, textual information, or edge type information when optimizing these embeddings.

#### Deep Learning

1. Define node neighborhoods
2. Talk about random walks 
3. Talk about auto encoders random walk independent approaches 

### Unifying Applications

Knowledge graphs have been used in many biomedical applications ranging from identifying protein functions [@doi:10.1186/s12859-018-2163-9] to prioritizing cancer genes [@doi:10.1093/bioinformatics/bty148] to recommending safer drugs to patients [@arxiv:1710.05980; @doi:10.1609/aaai.v33i01.33011126].
In this section we discuss how knowledge graphs are being applied in biomedical settings. 
We put particular emphasis on an emerging set of techniques: those that project knowledge graphs into a low dimensional space.

#### Genomic and Transcriptomic Applications

Knowledge graphs have been used to establish connections, such as associations or causal inference, between genes and diseases.
These connections can be established via gene prioritization, predicting novel edges between genes and dieases,and even analyzing RNA-seq analysis.
Through these tasks researchers have been able to generate new hypotheses to better understand disease.
In this section we discuss knowledge graphs being used in this context and highlight approaches that project these graphs into a low dimensional space to accomplish the task.
<font color=red>  Re-edit when you can come back to have better wording for above </font>

<font color=red> Matrix factorization on Networks for Genomic content </font>

10.1186/s12859-019-2956-5 <- MiRNA using Matrix decomp
^ matrix comp on MiRNA with Diseases
10.3390/genes10020080 <- MiRNA using custom matrix decomp 
^ matrix comp on MiRNA with Diseases
10.1109/TCBB.2019.2937774
^ More microRNA to Disease
10.1186/s12859-019-3260-0
^ All the MiRNA

<font color=red> Deep Learning Models for Networks within Genomic applications </font>

10.1093/nar/gkx750 - single cell RNA-SEQ
^ Node2vec with a bipartite network
10.1093/jamia/ocy117 - disease symptoms embeddings
^ Node to vec
10.1093/bioinformatics/btx275 - Neuro symbolic learning of knowledge graphs - used embeddings
^ Word2Vec variant
10.1093/bioinformatics/bty559 - smudge disease gene embeddings
^ More Node2vec variants
10.1186/s12920-019-0627-z
^ custom Node2Vec variant

10.1101/532226 - graph convolutional neural networks for gene prioritization
^ Graph convolutional networks

10.1109/BIBM47256.2019.8983134 <- Ironically the authors used an ensemble approach of matrix factorization as well as deep learning :mindblown:

<font color=red> Future direction: Translational Models for Networks on Genomic content </font>

#### Protein Protein Interactions

1. Mention predicting genes interacting genes

#### Drug Interactions

1. Talk about drug side effects
2. Drug repurposing
3. Drug-Disease Interations

#### Clinical applications

1. Can mention EHR use and other related applications
2. Mention Tiffany's work on private data embeddings
