## Applying Knowledge Graphs to Biomedical Challenges

1. Mention that these graphs can be used for discovery
2. Mention representation learning (aka representing a graph as dense vectors for nodes and/or edges)
3. 

### Unifying Techniques

1. Set up the problem that maps a knowledge graph into a low dimensional space

#### Matrix Factorization

1. Mention techniques for these with some papers

#### Deep Learning

1. Define node neighborhoods
2. Talk about random walks 
3. Talk about auto encoders random walk independent approaches 

### Unifying Applications

1. Mention how the previous section is used in a biomedical setting

#### Disease and Gene Interactions

1. Mention disease gene prioritization
2. Mention Disease gene associations

#### Protein Protein Interactions

1. Mention predicting genes interacting genes

#### Pharmaceutical Applications

Pharmaceutical applications consist of using knowledge graphs to discover new information on drugs.
Tasks in this context involve prediction drugs interacting with other drugs [@doi:10.1016/j.websem.2017.06.002], identifying molecular targets a drug might interact with [@doi:10.1155/2015/275045] and identifying new disease treatments for previously established drugs [@doi:10.7554/eLife.26726.001].
We discuss pharmaceutical applications that project knowledge graph onto a low dimensional space.

Similar to multi-omic applications knowledge graphs have been used in recommendation systems to infer novel information between drugs and diseases.
Dai et al. [@doi:10.1155/2015/275045] used collaborative filtering to infer drug-disease associations.
The authors constructed a drug-disease network by integrating two bipartite networks: a drug-gene interaction network and a disease-gene interaction network.
Ultimately most models utilize the popular word2vec variants to embed knolwedge graphs into a low dimensional space; however, translational models have been under utilized in these approaches which means people should work on incorporating more.
They integrated both networks under the assumption that a drug is associated with a disease if they interact with the same gene of interest. 
Then the authors generated an adjacency matrix where rows represent drugs and columns represent diseases and decomposed this matrix into two small rectangular matrices. 
These matrices were used to calculate similarity scores between all drugs and all diseases where high values implicate an association [@doi:10.1155/2015/275045].
Related approaches have been used to infer drug-target interactions [@doi:10.1109/TCBB.2016.2530062; @doi:10.1109/BIOCAS.2018.8584817; @doi:10.1093/bioinformatics/btaa010] and drug-disease treatments [@doi:10.1109/TCBB.2018.2830384; @doi:10.1371/journal.pcbi.1004760; @doi:10.1038/srep40376; @doi:10.1021/ci500340n; @doi:10.1186/s12859-018-2220-4]
In spite of reported success these approaches are limited to the drugs and diseases contained in the network, which makes predicting unseen drugs impractical.

Deep learning techniques consist of using node2vec-like models [@doi:10.1093/bioinformatics/btx160; @doi:10.1101/539643] and auto-encoders [@arxiv:1804.10850v1; @arxiv:1802.00543] to project knowledge graphs into a low dimensional space.
Zong et al. [@doi:10.1093/bioinformatics/btx160] used a node2vec-like model to predict drug-target associations.
The authors constructed a disease-target-disease network using drug centered databases: Drugbank [@doi:10.1093/nar/gkm958] and Diseasome [@doi:10.1073/pnas.0701361104].
After constructed, the authors used a random walk on the network to determine how nodes are interconnected with one another and then trained a skip-gram model to generate node embeddings.
Lastly, the authors constructed a similairty metric to rank how similar drugs are to their targets [@doi:10.1093/bioinformatics/btx160].
A limitation to this method is that their network is missing information such as pharmacological class or drug checmical structure that may help in prediciton performacne.
Overall, deep learning is a robust set of techniques that perform well when undergoing pharmaceutical applications and has been shown to outperform most linear approaches [@doi:10.1186/s12859-019-3284-5; @doi:10.1145/3307339.3342161].

Knowledge graphs have aided the pharmaceutical field in by predicting drug information such as drug side effects and new disease treatments.
Majority of methods have been heavily focused on matrix factorization and deep learning techniques.
Due to the success of deep learning [@doi:10.1186/s12859-019-3284-5; @doi:10.1145/3307339.3342161] the focus of the field has shifted to these techniques; however, a possible extension is to use an ensemble of deep learning techniques and linear methods to improve performance.
Plus, another area of exploration is to incorporate information such as phamaceutical classes for drugs or chemical structure to improve knowledge graph detection power.

#### Clinical applications

1. Can mention EHR use and other related applications
2. Mention Tiffany's work on private data embeddings
