## Applying Knowledge Graphs to Biomedical Challenges

1. Mention that these graphs can be used for discovery
2. Mention representation learning (aka representing a graph as dense vectors for nodes and/or edges)
3. 

### Unifying Techniques

1. Set up the problem that maps a knowledge graph into a low dimensional space

#### Matrix Factorization

1. Mention techniques for these with some papers

#### Deep Learning

1. Define node neighborhoods
2. Talk about random walks 
3. Talk about auto encoders random walk independent approaches 

### Unifying Applications

1. Mention how the previous section is used in a biomedical setting

#### Disease and Gene Interactions

1. Mention disease gene prioritization
2. Mention Disease gene associations

#### Protein Protein Interactions

1. Mention predicting genes interacting genes

#### Pharmaceutical Applications

Pharmaceutical applications consist of using resources such as knowledge graphs to discover essental information on drugs.
Usually this information involves understanding the mechanisms of one drug interacting with another [@doi:10.1016/j.websem.2017.06.002], molecular targets a drug may focus on [@doi:10.1155/2015/275045] and identifying novel diseases previously established drugs may treat [@doi:10.7554/eLife.26726.001].
We discuss applications that utilize knowledge graph embeddings to perform pharmaceutical applications.

Similar to multi-omic applications knowledge graphs have been used in recommendation systems to infer novel information between drugs and diseases.
Dai et al. [@doi:10.1155/2015/275045] used collaborative filtering to infer drug disease associations.
The authors constructed a drug-disease network by integrating two bipartite networks: a drug-gene interaction network and a disease-gene interaction network.
They integrated both networks under the assumption that a drug is associated with a disease given they share the same gene of interest. 
Upon construction of a drug-diease simialrity matrix (equivalent to an adjacency matrix), the authors used matrix factorization to decompose it into two small rectangular matricies. 
Using these two small matricies the authors were reconstructed a similarity matrix between all drugs and all diseases where high values implicate an association between drugs and diseases [@doi:10.1155/2015/275045].
Approaches akin to above have been used to infer drug-target interactions [@doi:10.1109/TCBB.2016.2530062; @doi:10.1109/BIOCAS.2018.8584817; @doi:10.1093/bioinformatics/btaa010] and drug-disease interactions [@doi:10.1109/TCBB.2018.2830384; @doi:10.1371/journal.pcbi.1004760; @doi:10.1038/srep40376; @doi:10.1021/ci500340n; @doi:10.1186/s12859-018-2220-4]
Inspite of reported success of these approaches, validation was still a big challenge and remains an open project for future techniques to consider.

Approaches that use deep learning involve the use of node2vec, autoencoders and variants of both.
Zong et al. [@doi:10.1093/bioinformatics/btx160] used a node2vec-like technique to predict drug-target associations.
The authors used drug centered databases, Drugbank [@doi:10.1093/nar/gkm958] and Diseasome [@doi:10.1073/pnas.0701361104], to construct a drug-target-disease network.
After constructing the network, they authors used a random walk on the network to gather contextual information on how nodes are connected with one another and then trained a skip-gram generate node embeddings.
Lastly the authors used a similarity based metric to rank how similar drugs were to their targets [@doi:10.1093/bioinformatics/btx160].
A variant of this approach was used to infer drug-target connections [@doi:10.1101/539643].
Besides node2vec autoencoders were used to predict drug similairty [@arxiv:1804.10850v1], infer drug side-effects [@arxiv:1802.00543] and make drug target preductions [@doi:10.1089/cmb.2017.0135].
Overall deep learning is a robust set of techniques that perform well when undergoing pharmaceutical applications.
It has been shown that deep learning outperforms most linear approaches [@doi:10.1186/s12859-019-3284-5; @doi:10.1145/3307339.3342161], which is where majority of this ifield is headed.

Ultimately most models utilize the popular word2vec variants to embed knolwedge graphs into a low dimensional space; however, translational models have been under utilized in these approaches which means people should work on incorporating more.
Plus this area is still premature as many of these papers were published in 2019, which means there is more room for improvement.

#### Clinical applications

1. Can mention EHR use and other related applications
2. Mention Tiffany's work on private data embeddings
