## Applying Knowledge Graphs to Biomedical Challenges

1. Mention that these graphs can be used for discovery
2. Mention representation learning (aka representing a graph as dense vectors for nodes and/or edges)
3. 

### Unifying Techniques

Mapping high dimensional data into a low dimensional space has greatly improved modeling performance in fields such as natural language processing [@arxiv:1301.3781; @arxiv:1310.4546] and image analysis [@arxiv:1512.03385].
The success of these approaches provides rationale for projecting knowledge graphs into a low dimensional space as well [@arxiv:1709.05584].
Techniques that perform this projection often require information on how nodes are connected with one another [@raw:gongreplearn; @doi:10.1109/TKDE.2016.2606098; @raw:hanreplearn; @doi:10.1145/2806416.2806512], while other approaches can work directly with the edges themselves [@arxiv:1610.04073].
We group methods for producing low-dimensional representations of knowledge graphs into the following three categories: matrix factorization, translational methods, and deep learning.

#### Matrix Factorization

1. Mention techniques for these with some papers

#### Translational Distance Models

Translational distance models treat edges in a knowledge graph as linear transformations.
As an example, one such algorithm, TransE [@raw:bordestranse], treats every node-edge pair as a triplet with head nodes represented as $\textbf{h}$, edges represented as $\textbf{r}$, and tail nodes represented as $\textbf{t}$.
These representations are combined into an equation that mimics the iconic word vectors translations ($\textbf{king} - \textbf{man} + \textbf{woman} \approx \textbf{queen}$) from the Word2vec model [@arxiv:1310.4546].
The equation is shown as follows: $\textbf{h} + \textbf{r} \approx \textbf{t}$.
Starting at the head node ($\textbf{h}$), add the edge vector ($\textbf{r}$) and the result should be the tail node ($\textbf{t}$).
TransE optimizes embeddings for $\textbf{h}$, $\textbf{r}$, $\textbf{t}$, while guaranteeing the global equation ($\textbf{h} + \textbf{r} \approx \textbf{t}$) is satisfied [@raw:bordestranse].
A caveat to the TransE approach is that it the training steps force relationships to have a one to one mapping, which may not be appropriate for all types of relationships.

Wang et al. [@raw:wangtransH] attempted to resolve the one to one mapping issue by developing the TransH model.
TransH treats relations as hyperplanes rather than a regular vector and projects the head ($\textbf{h}$) and tail ($\textbf{t}$) nodes onto the hyperplane.
Following this projection, a distance vector ($\textbf{d}_{r}$) is calculated between the projected head and tail nodes.
Finally, each vector is optimized while preserving the global equation ($\textbf{h} + \textbf{d}_{r} \approx \textbf{t}$) [@raw:wangtransH].
Other approaches [@raw:lintransR, @arxiv:1610.04073; @arxiv:1909.00672] have built off of the TransE and TransH models. 
In the future, it may be beneficial for these models is to incorporate other types of information such as edge confidence scores, textual information, or edge type information when optimizing these embeddings.

#### Deep Learning

1. Define node neighborhoods
2. Talk about random walks 
3. Talk about auto encoders random walk independent approaches 

### Unifying Applications

Knowledge graphs have been used in many biomedical applications ranging from identifying protein functions [@doi:10.1186/s12859-018-2163-9] to prioritizing cancer genes [@doi:10.1093/bioinformatics/bty148] to recommending safer drugs to patients [@arxiv:1710.05980; @doi:10.1609/aaai.v33i01.33011126].
In this section we discuss how knowledge graphs are being applied in biomedical settings. 
We put particular emphasis on an emerging set of techniques: those that project knowledge graphs into a low dimensional space.

<font color=red> Matrix factorization on Networks for Genomic content </font>

#### Multi-Omic Applications

Multi-omic applications consist of analyzing genes (genomics), RNA (transcriptomics) and proteins (proteomics).
Knowledge graphs have been used in this area to establish connections between the biological entities themselves (DNA, RNA, proteins) and with diseases.
These connections are established via applications such as gene prioritization [@doi:10.1093/jamia/ocy117], predicting interactions between proteins [@doi:10.1089/cmb.2012.0273;10.1186/1752-0509-9-S1-S9], and even classifying MiRNA-disease associations [@doi:10.1155/2017/2498957].
In this section we discuss approaches that project knowledge graphs into a low dimensional space under the context of multi-omic applications.

Knowledge graphs have been used as recommendation systems to make novel discoveries.
Formally known as collaborative filtering [@doi:10.1155/2009/421425], these systems use matrix factorization techniques on an adjacency matrix to generate knowledge graph embeddings.
Then these embeddings are used to calcuate similarity scores between nodes.
Shen et al. [@doi:10.1155/2017/2498957] used collaborative filtering to establish an assocation between MiRNA and diseases.
The authors constructed an MiRNA-Disease network using the HMDDD database [@doi:10.1093/nar/gky1010] and generated an adjacency matrix with the rows representing MiRNA and the columns represent the diseases.
The adjacency matrix was decomposed into small rectangular matricies using SVD and similarity scores were calculated between the MiRNA and diseases where higher scores implicate a high liklihood of association [@doi:10.1155/2017/2498957].
Other appraoches have built off of Shet et al.'s [@doi:10.1155/2017/2498957] work by integrating extra network features [@doi:10.3390/genes10020080;@doi:10.1186/s12859-019-2956-5;@doi:10.1186/s12859-019-3260-0] or integraing machine learning models [@doi:10.1109/TCBB.2019.2937774].
Besides MiRNA collaborative filtering has been used to predict protein-protein interactions [@doi:10.1109/BIBM.2010.5706537; @doi:10.1089/cmb.2012.0273; @doi"10.1186/1752-0509-9-S1-S9].
These methods only one to two different data types; however, an extension to these methods is to incorporate other data types like genes or protein mutations to increases detection power.

Deep learning approaches in this context have mainly focused on the node2vec model  [@arxiv:1607.00653] or variants of the method.
Yang et al. [@doi:10.1093/jamia/ocy117] used node2vec to estabish a recommendation system that calculates similarity between genes and disease symptoms.
The authors constructed a symptom-gene knowledge graph by connecting genes to symptoms via the diseases both the gene and symptom may share.
Following the graph construction, the generated knowledge was embedded via node2vec and scores for calculate for every gene-symptom in the knoweldge graph [@doi:10.1093/jamia/ocy117].
This appraoch outperformed non knowldge based methods; however, a direction that could be taken is integrating experimental data such as gene or protein expressoin to improve performance.
Besides gene-symptoms other approaches used knowledge graphs to predict gene-disease associations
[@doi:10.1093/bioinformatics/bty559; doi:10.3389/fgene.2019.00226 @doi:10.1186/s12920-019-0627-z; @doi:10.1109/BIBM47256.2019.8983134] analyze RNA-seq data [@doi:10.1093/nar/gkx750] and predict protein protein infromation [@doi:10.1093/bioinformatics/btx275; @doi:10.1007/978-3-030-00825-3_16; @doi:10.1016/j.artmed.2019.04.001; @doi:10.1186/s12859-018-2163-9].

Knowledge graph embeddings have helped the multi-omics field discover novel information.
These methods have been heavily used in constructing recommendation systems to construct connections between diseases and genes, RNA and diseases and protein-protein interactions.
Despite reported success many of these methods have focused on using node2vec variants and other deep learning techinques; however, a set of techniques that haven't been used is translational models. 
Plus, this field would benefit from incorporating experimental data such as gene expression and epigenetic data 

#### Drug Interactions

1. Talk about drug side effects
2. Drug repurposing
3. Drug-Disease Interations

#### Clinical applications

1. Can mention EHR use and other related applications
2. Mention Tiffany's work on private data embeddings
